spec/unit/puppet/type/dsc_package_spec.rb

```diff
@@ -202,12 +202,14 @@
     expect{dsc_package[:dsc_returncode] = true}.to raise_error(Puppet::ResourceError)
   end
 
-  it 'should not accept int for dsc_returncode' do
-    expect{dsc_package[:dsc_returncode] = -16}.to raise_error(Puppet::ResourceError)
+  it 'should accept int for dsc_returncode' do
+    dsc_package[:dsc_returncode] = [32, 64, 128]
+    expect(dsc_package[:dsc_returncode]).to eq([32, 64, 128])
   end
 
-  it 'should not accept uint for dsc_returncode' do
-    expect{dsc_package[:dsc_returncode] = 16}.to raise_error(Puppet::ResourceError)
+  it 'should accept uint for dsc_returncode' do
+    dsc_package[:dsc_returncode] = [32, 64, 128]
+    expect(dsc_package[:dsc_returncode]).to eq([32, 64, 128])
```

Comment:
Something doesn't seem right here.  It should accept `int` and it should accept `uint`?

The MOF reads like:

```
[write] uint32 ReturnCode[];
```

I think there's a bug with the test generators.